### DISCUSSION TITLE
Z-Wave UZB-7 - 7.21.0 firmware - HA device inclusion fails

### DISCUSSION THREAD

**Discussion by jonr6846** [zwave-js/zwave-js#6994] [Q&A]
I'm running the 2024-07 version of Home Assistant with Z-Wave JS UI 9.14.4.40080a1 and Z-Wave JS 12.12.1
Inclusion worked OK until I upgraded the UZB-7 stick to firmware version 7.21.0, after which inclusion fails on my Shelly devices with no more useful information in the logs that inclusion failed, even after factory resets on all devices/controller

Moving those devices/controller to to a Win10 PC running Simplicity Studio, they include and work fine - The problem is only evident using HA and Z-Wave JS UI

I can't find a newer firmware version than 7.21.0; I'm not sure it exists

Has the Firmware update to 7.21.0 broken the system?
Thanks

**Comment by AlCalzone** [comment]
There is a 7.21.3, see https://github.com/SiliconLabs/gecko_sdk/tree/v4.4.3/protocol/z-wave/Apps/bin/gbl how to find the file

**Reply by jonr6846** [reply]
@AlCalzone - Thanks for that - not knowing git lfs, I got a friend to capture it and it loaded onto the UZB-7 stick OK but the inclusion problem persists  ...
![image](https://github.com/zwave-js/node-zwave-js/assets/102229005/b41b323e-6ab4-4439-b0b2-3be8962c825b)

**Reply by AlCalzone** [reply]
Please make a [driver log](https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=driver-logs) of an inclusion attempt, loglevel `debug` and attach it here as a file (drag & drop into the text field).

Also how close can you bring the device to the controller? Just for testing.

**Comment by jonr6846** [comment]
[zwave-js-ui-store (2).zip](https://github.com/user-attachments/files/16162281/zwave-js-ui-store.2.zip)
The Z-Wave controller and Shelly Qubino Wave 1 mini and physically with 2 meters of each other (across the desk)
Thanks for taking a look at this

**Reply by AlCalzone** [reply]
> 2024-07-10T11:29:46.511Z CNTRLR   handling add node request (status = NodeFound)
> 2024-07-10T11:29:49.042Z CNTRLR   handling add node request (status = Failed)

So the node is found, but there is some issue during communication. Unfortunately this is all Z-Wave JS gets told, the rest is between the stick and the device.

The fact that it works on a different machine leads me to believe you have an interference problem. If you haven't already, put the stick on a short USB extension cable to get it away from USB ports. [This video](https://www.youtube.com/watch?v=tHqZhNcFEvA) demonstrates the problem for Zigbee, but it is just the same for Z-Wave.
FWIW, I can plug Z-Wave sticks into the front panel of my desktop PC just fine, but they don't work when plugged into my laptop.

**Comment by jonr6846** [comment]
Thanks for your response but I've had no success with this combination. I've created a new clean install 6/7 times away from WiFi/mobiles or other potential interference, the dongle has always been on a shielded USB extension. For me the inclusion and commands work perfectly using Simplicity Studio (SS) but not using HA and zwave js iu . I've spent a world of time and quite a bit of money on this and for me I'll have to move on and stick with zigbee which is working well. I have massive respect for all those participated in this development and thank you for all you've done.
Kind regards
P.S. Using SS the device reset is instant but in HA, the device reset screen hangs - It looks like HA is expecting a reply that never comes.

**Reply by AlCalzone** [reply]
> For me the inclusion and commands work perfectly using Simplicity Studio (SS) but not using HA and zwave js iu

Have you tried using Z-Wave JS UI on the same machine, the same USB port as SS?
